presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/ObjectBlockPositionStateSerializer.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.function.AccumulatorStateSerializer;
+import com.facebook.presto.spi.type.Type;
+
+public class ObjectBlockPositionStateSerializer
+        implements AccumulatorStateSerializer<ObjectBlockPositionState>
+{
+    private final Type type;
+
+    public ObjectBlockPositionStateSerializer(Type type)
+    {
+        this.type = type;
+    }
+
+    @Override
+    public Type getSerializedType()
+    {
+        return type;
+    }
+
+    @Override
+    public void serialize(ObjectBlockPositionState state, BlockBuilder out)
+    {
+        if (state.getBlock() == null) {
+            out.appendNull();
+        }
+        else {
+            type.writeObject(out, type.getObject(state.getBlock(), state.getPosition()));
+        }
+    }
+
+    @Override
+    public void deserialize(Block block, int index, ObjectBlockPositionState state)
```

Comment:
Add comments explain for the deserialized states, the indices will be consecutive and the block is compacted.